x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processors/processor_field_selector.tsx

```diff
@@ -22,28 +31,70 @@ export const ProcessorFieldSelector = () => {
   const { invalid, error } = fieldState;
 
+  const isUnsupported = unsupportedFields.some((unsupportedField) =>
+    value.startsWith(unsupportedField)
+  );
+
   return (
-    <EuiFormRow
-      label={i18n.translate(
-        'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorLabel',
-        { defaultMessage: 'Field' }
-      )}
-      helpText={i18n.translate(
-        'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorHelpText',
-        { defaultMessage: 'Field to search for matches.' }
-      )}
-      isInvalid={invalid}
-      error={error?.message}
-    >
-      <EuiFieldText
-        data-test-subj="streamsAppProcessorFieldSelectorFieldText"
-        {...inputProps}
-        inputRef={ref}
+    <>
+      <EuiFormRow
+        label={i18n.translate(
+          'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorLabel',
+          { defaultMessage: 'Field' }
+        )}
+        helpText={i18n.translate(
+          'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorHelpText',
+          { defaultMessage: 'Field to search for matches.' }
+        )}
         isInvalid={invalid}
-      />
-    </EuiFormRow>
+        error={error?.message}
+      >
+        <EuiFieldText
+          data-test-subj="streamsAppProcessorFieldSelectorFieldText"
+          {...inputProps}
+          value={value}
+          inputRef={ref}
+          isInvalid={invalid}
+        />
+      </EuiFormRow>
+      {isUnsupported && (
+        <EuiCallOut
+          color="warning"
+          iconType="alert"
+          title={i18n.translate(
+            'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorUnsupportedDottedFieldsWarning.title',
+            {
+              defaultMessage: 'Dot-separated field names are not supported for processors.',
+            }
+          )}
+          css={css`
+            margin-top: ${euiTheme.size.s};
+            margin-bottom: ${euiTheme.size.m};
+          `}
+        >
+          <p>
+            {i18n.translate(
+              'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorUnsupportedDottedFieldsWarning.p1',
+              {
+                defaultMessage:
+                  'Using field names with dots may lead to misleading simulation results, our tool does not support this scenario.',
+              }
+            )}
+          </p>
+          <p>
+            {i18n.translate(
+              'xpack.streams.streamDetailView.managementTab.enrichment.processor.fieldSelectorUnsupportedDottedFieldsWarning.p2',
+              {
+                defaultMessage:
+                  'We are working to improve support for such cases. For accurate results, consider avoiding dotted field names or expanding them into nested objects until this feature is fully implemented.',
+              }
+            )}
+          </p>
+        </EuiCallOut>
```

Comment:
@mdbirnstiehl could you please take a look at these copies and the wording? Let me know if you need more context about it.